### PR TITLE
Created eventuate common quarkus modules.

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/DataSourceSelectorConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import javax.sql.DataSource;
+
+@ApplicationScoped
+@Alternative
+@Priority(1)
+public class DataSourceSelectorConfiguration {
+
+  @Produces
```
Comment (cer) [inline eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/DataSourceSelectorConfiguration.java]: As discussed, this is test code. An Eventuate application should not be required to use this.

File: eventuate-common-quarkus-jdbc/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,17 @@
+quarkus.datasource.mysql.db-kind=mysql
```
Comment (cer) [inline eventuate-common-quarkus-jdbc/src/test/resources/application.properties]: As discussed, this is test code.

File: eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import java.util.Optional;
+
+@ApplicationScoped
+public class IdGeneratorConfiguration {
+  @Produces
+  public IdGenerator idGenerator(@ConfigProperty(name = "eventuate.outbox.id") Optional<Long> id) {
+    return id
```
Comment (cer) [inline eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java:16]: Is this code that is duplicated elsewhere?

File: eventuate-common-quarkus-in-memory-database/src/main/java/io/eventuate/common/quarkus/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import javax.sql.DataSource;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Alternative
+@Priority(0)
+public class EventuateCommonInMemoryDatabaseConfiguration {
+  @Produces
+  public DataSource dataSource(Instance<EventuateDatabaseScriptSupplier> scripts) {
+    return new EventuateInMemoryDataSourceBuilder(scripts.stream().collect(Collectors.toList())).build();
```
Comment (cer) [inline eventuate-common-quarkus-in-memory-database/src/main/java/io/eventuate/common/quarkus/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java:20]: Is this code that's duplicated elsewhere?

File: eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/EventuateCommonJdbcOperationsConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  @Produces
+  public EventuateCommonJdbcOperations eventuateCommonJdbcOperations(EventuateJdbcStatementExecutor eventuateJdbcStatementExecutor,
+                                                                     SqlDialectSelector sqlDialectSelector,
+                                                                     @ConfigProperty(name = "eventuateDatabase") String dbName) {
+    return new EventuateCommonJdbcOperations(eventuateJdbcStatementExecutor, sqlDialectSelector.getDialect(dbName, Optional.empty()));
+  }
+
+  @Produces
+  public EventuateJdbcStatementExecutor eventuateJdbcStatementExecutor(Instance<DataSource> dataSource) {
+    return new EventuateCommonJdbcStatementExecutor(() -> {
```
Comment (cer) [inline eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/EventuateCommonJdbcOperationsConfiguration.java:33]: Is this code that is duplicated elsewhere?

**Review by cer — CHANGES_REQUESTED** [review]

File: eventuate-common-quarkus-jdbc/src/test/resources/application.properties (hunk 2 above)
Comment (dartartem) [reply eventuate-common-quarkus-jdbc/src/test/resources/application.properties]: yes, I will fix

File: eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java (hunk 3 above)
Comment (dartartem) [reply eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java:16]: Could you please clarify, what do you mean?

Comment (cer) [reply eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java:16]: Do these four lines of code also occur in eventuate-common?

```
id
            .map(DatabaseIdGenerator::new)
            .map(generator -> (IdGenerator)generator)
            .orElseGet(ApplicationIdGenerator::new);
```

Comment (dartartem) [reply eventuate-common-quarkus-id/src/main/java/io/eventuate/common/quarkus/id/IdGeneratorConfiguration.java:16]: No, it looks different for micronaut and spring.
spring: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-spring-id/src/main/java/io/eventuate/common/spring/id/IdGeneratorConfiguration.java
micronaut: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-id/src/main/java/io/eventuate/common/micronaut/id/IdGeneratorFactory.java

File: eventuate-common-quarkus-in-memory-database/src/main/java/io/eventuate/common/quarkus/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java (hunk 4 above)
Comment (dartartem) [reply eventuate-common-quarkus-in-memory-database/src/main/java/io/eventuate/common/quarkus/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java:20]: For micronaut/spring in is different.
micronaut: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-in-memory-database/src/main/java/io/eventuate/common/micronaut/inmemorydatabase/EventuateCommonInMemoryDatabaseFactory.java
spring: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-spring-in-memory-database/src/main/java/io/eventuate/common/spring/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java

File: eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/EventuateCommonJdbcOperationsConfiguration.java (hunk 5 above)
Comment (dartartem) [reply eventuate-common-quarkus-jdbc/src/main/java/io/eventuate/common/quarkus/jdbc/EventuateCommonJdbcOperationsConfiguration.java:33]: No. Spring: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-spring-jdbc/src/main/java/io/eventuate/common/spring/jdbc/EventuateCommonJdbcOperationsConfiguration.java#L24
Micronaut: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-spring-jdbc/src/main/java/io/eventuate/common/micronaut/spring/jdbc/EventuateJdbcMicronautSpringFactory.java#L21
and https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-data-jdbc/src/main/java/io/eventuate/common/micronaut/data/jdbc/EventuateJdbcMicronautDataFactory.java#L16

**Pull request merged**